examples/egui/main.rs

```diff
@@ -0,0 +1,48 @@
+mod editor;
+mod fox;
+
+use dotrix::{
+    Dotrix,
+    ecs::{ Mut, RunLevel, System },
+    input::{ Mapper },
+    math::{ Point3 },
+    renderer::{ Light },
+    services::{ Assets, Camera, Frame, Input, World },
+    systems::{ overlay_update, skeletal_animation, world_renderer },
+};
+use editor::{ Editor };
+
+#[derive(PartialEq, Eq, Hash, Copy, Clone)]
+pub enum Action {}
+
+fn main() {
+    Dotrix::application("egui Example")
+        .with_system(System::from(spawn_lights).with(RunLevel::Startup))
+        .with_system(System::from(fox::startup).with(RunLevel::Startup))
+        .with_system(System::from(skeletal_animation))
+        .with_system(System::from(editor::startup).with(RunLevel::Startup))
+        .with_system(System::from(overlay_update))
+        .with_system(System::from(editor::ui))
+        .with_system(System::from(editor::update_camera))
+        .with_system(System::from(editor::update_fox))
+        .with_system(System::from(editor::update_lights))
+        .with_system(System::from(world_renderer).with(RunLevel::Render))
+        .with_service(Assets::new())
+        .with_service(Frame::new())
+        .with_service(Editor::new())
+        .with_service(Camera {
+            distance: 222.0,
+            y_angle: 0.74,
+            xz_angle: 0.25,
+            target: Point3::new(0.0, 0.5, 0.0),
+            ..Default::default()
+        })
+        .with_service(World::new())
+        .with_service(Input::new(Box::new(Mapper::<Action>::new())))
+        .run();
+}
+
+pub fn spawn_lights(mut world: Mut<World>) {
+    world.spawn(Some((Light::white([0.0, 500.0, 0.0]),)));
+    world.spawn(Some((Light::white([200.0, 50.0, 200.0]),)));
```

Comment:
[info] I've never tested multiple light sources before, but it seems it works :)

Reply:
Yes, I experimented a bit with it. I also wanted to have separated controls for lights with disabling checkoxes, but when I added `pub enabled: bool,` into `dotrix_core\src\renderer\light.rs`, it stopped working correctly.

Reply:
This is actually could be a usefull feature, could you please add an issue for this? As a feature request or something like this.

Reply:
Yes, I will add